### PR TITLE
Add ToC to SPECIFICATION.md

### PR DESCRIPTION
**Pull request by dbrgn**
I think this aids navigation.

### CODE REVIEW

**Comment by dbrgn** [comment]
I also reorganized the sections a bit.

File: SPECIFICATION.md [hunk 1]
```diff
@@ -15,7 +25,34 @@ quadratic / cubic curves.
...
+axis (X=0). Glyphs that are not aligned to the Y axis have an impact on letter
+spacing, see section "Letter Spacing" for details.
+
+Following image illustrates where the baseline (Y=0) and cap height (Y=9) are
+located:
+
+![glyph geometry](images/glyph_geometry.png)
+
+
+## <a name="highlevel"></a> 3. Format Description
```
Comment (ubruhin) [inline SPECIFICATION.md]: Anchor ("highlevel") and title ("Format Description") aren't consistent.

File: SPECIFICATION.md [hunk 2]
```diff
@@ -96,7 +133,7 @@ three things MUST be as following:
 3. 0-1 whitespace definitions (e.g. `~3.6`)
 
 
-#### Polylines
+#### 2.2.1. Polylines
```
Comment (ubruhin) [inline SPECIFICATION.md]: Must be 3.2.1

File: SPECIFICATION.md [hunk 3]
```diff
@@ -116,7 +153,7 @@ LETTER A (U+0041)":
     0.8333,2.5;5.1666,2.5
     0,0;3,9;6,0
 
-##### Circular Arc Segments
+##### 2.2.1.1. Circular Arc Segments
```
Comment (ubruhin) [inline SPECIFICATION.md]: 3.2.1.1.

File: SPECIFICATION.md [hunk 4]
```diff
@@ -135,7 +172,7 @@ Example:
     1,0;1,7.5,-4.5;2.5,9;3,9
     0,6;3,6
 
-#### References
+#### 2.2.2. References
```
Comment (ubruhin) [inline SPECIFICATION.md]: 3.2.2.

**Review by ubruhin — CHANGES_REQUESTED** [review]

**Comment by dbrgn** [comment]
Good catch. Should be fixed.

**Review by ubruhin — APPROVED** [review]
:+1:

**Pull request merged**